### PR TITLE
File Based Analytics for Microgateway

### PR DESCRIPTION
**Pull request by fazlan-nazeem**
## Purpose

This PR implements analytics support for microgateway. 

This includes a JAX-RS API for publishing analytics data and a custom event receiver to read those files.

## Documentation
https://docs.google.com/document/d/1Oaope3eb7W4O7afU_ILFRIKsJJhNvojBEJQNryXh3tA/edit

### CODE REVIEW

File: components/org.wso2.analytics.apim.file.adapter/src/main/java/org/wso2/analytics/apim/file/adapter/FileEventAdapter.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    }
+
+    @Override
+    public void destroy() {
+
+    }
+
+    @Override
+    public boolean isEventDuplicatedInCluster() {
+        return Boolean.parseBoolean(eventAdapterConfiguration.getProperties().get(EventAdapterConstants.EVENTS_DUPLICATED_IN_CLUSTER));
```
Comment (ruks) [inline components/org.wso2.analytics.apim.file.adapter/src/main/java/org/wso2/analytics/apim/file/adapter/FileEventAdapter.java]: formatting error?

File: components/org.wso2.analytics.apim.file.adapter/src/main/java/org/wso2/analytics/apim/file/adapter/internal/ds/FileEventAdapterServiceValueHolder.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+* limitations under the License.
+*/
+package org.wso2.analytics.apim.file.adapter.internal.ds;
+
+import org.wso2.analytics.apim.file.adapter.FileEventAdapter;
+import org.wso2.carbon.event.stream.core.EventStreamService;
+
+import java.util.concurrent.ConcurrentHashMap;
+
+public final class FileEventAdapterServiceValueHolder {
```
Comment (ruks) [inline components/org.wso2.analytics.apim.file.adapter/src/main/java/org/wso2/analytics/apim/file/adapter/internal/ds/FileEventAdapterServiceValueHolder.java]: seems formatting errors in the class

Comment (fazlan-nazeem) [reply components/org.wso2.analytics.apim.file.adapter/src/main/java/org/wso2/analytics/apim/file/adapter/internal/ds/FileEventAdapterServiceValueHolder.java]: This is not getting changed when I do a formatting via the IDE.

Comment (fazlan-nazeem) [reply components/org.wso2.analytics.apim.file.adapter/src/main/java/org/wso2/analytics/apim/file/adapter/internal/ds/FileEventAdapterServiceValueHolder.java]: made the changes manually

**Pull request merged**